core/include/vecmem/containers/data/jagged_vector_buffer.hpp

```diff
@@ -79,7 +79,8 @@ class jagged_vector_buffer : public jagged_vector_view<TYPE> {
     ///        be host accessible.
     /// @param host_access_resource An optional host accessible memory
     ///        resource. Needed if @c resource is not host accessible.
-    jagged_vector_buffer(const std::vector<std::size_t>& sizes,
+    template <typename size_t = std::size_t>
```

Comment:
Can we name this `size_type`? `size_t` is too easily confused with `std::size_t`.